cfme/utils/version.py

```diff
@@ -144,6 +144,15 @@ def pick(v_dict, active_version=None):
     return v_dict.get(sorted_matching_versions[0]) if sorted_matching_versions else None
 
 
+def new_pick(v_dict, appliance):
+    """Does the same as ``pick``, but requires an appliance as an argument."""
```

Comment:
**required** lets put the logic that takes the active version into a own function

the new_pick that takes the appliance and the old pick can then share most code